src/platforms/esp32/components/avm_sys/sys.c

```diff
@@ -759,3 +785,76 @@ term esp_err_to_term(GlobalContext *glb, esp_err_t status)
             return term_from_int(status);
     }
 }
+
+int sys_mbedtls_entropy_func(void *entropy, unsigned char *buf, size_t size)
+{
+#ifndef MBEDTLS_THREADING_C
```

Comment:
In fact, if we are building for multicore, we should define MBEDTLS_THREADING_C on esp32 & pico for a finer-grained locking directly managed by mbedtls (they know when to lock and unlock).
On esp32, it's disabled by default but it's part of the sdk config.
On Pico, we provide our own mbedtls-config.h, so we can just add it there.

Then we need to define MBEDTLS_THREADING_ALT and call mbedtls_threading_set_alt.
On esp32, this is defined by default.
On Pico, we need to add it to mbedtls-config.h as well.

On generic unix, we don't control how mbedtls was compiled, so it makes sense to keep this logic and handle the locking ourselves.

Reply:
I did a quick check and they don't look much more fine grained than this implementation.
On ESP32 I didn't want to alter the default configuration and I wanted to support both scenarios, since supporting them didn't cost a large effort I just implemented that wrapper function and handled both configurations.
Right now we customize sdkconfig just for partitioning and that's it.

Reply:
Right.

Still we probably want to use mbedtls native threading on Pico as we're providing our own config file.